lib/std/os.zig

```diff
@@ -1275,6 +1274,41 @@ pub const MakeDirError = error{
     BadPathName,
 } || UnexpectedError;
 
+pub fn mkdirat(dir_fd: fd_t, sub_dir_path: []const u8, mode: u32) MakeDirError!void {
+    if (builtin.os == .windows) {
+        const sub_dir_path_w = try windows.sliceToPrefixedFileW(sub_dir_path);
+        @compileError("TODO implement mkdirat for Windows");
```

Comment:
Looks like NtCreateFile supports creating directories, so this should be fairly straightforward to implement. Windows will even let us implement `makeOpenDir` and `makeOpenPath` with atomic semantics. For POSIX, we'll have to make them non-atomic and document that fact in the doc comments. See the related use case in lib/std/build/write_file.zig where it says "replace with something like fs.makePathAndOpenDir"

See https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/nf-ntifs-ntcreatefile and the openFileWindows function.

Reply:
I've already got an implementation locally; just need to clean up and add to this PR :)